File: src/main/java/sample/MemberInfoPrinter.java
package sample;

public class MemberInfoPrinter {
    private MemberDao memDao;
    private MemberPrinter printer;

    public void setMemberDao(MemberDao memberDao) {
        this.memDao = memberDao;
    }

    public void setPrinter(MemberPrinter printer) {
        this.printer = printer;
    }

    public void printMemberInfo(String id) {
        Member member = memDao.selectByLoginId(id);
        if (member == null) {
            System.out.println("nodata");
            return;
        }
        printer.print(member);
        System.out.println();
    }
}

File: src/main/java/sample/MemberInfoPrinter.kt
package sample

import org.springframework.beans.factory.annotation.Autowired

class MemberInfoPrinter {
    @Autowired
    private var memDao: MemberDao? = null
    private var printer: MemberPrinter? = null
    fun setMemberDao(memberDao: MemberDao?) {
        memDao = memberDao
    }

    @Autowired
    fun setPrinter(printer: MemberPrinter?) {
        this.printer = printer
    }

    fun printMemberInfo(id: String?) {
        val member = memDao!!.selectByLoginId(id)
        if (member == null) {
            println("nodata")
            return
        }
        printer!!.print(member)
        println()
    }
}
